src/Microsoft.DotNet.ProjectModel/Files/IncludeContext.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.ProjectModel.Files
+{
+    public class IncludeContext
+    {
+        private static readonly char[] PatternSeparator = new[] { ';' };
+
+        public IncludeContext(
+            string sourceBasePath,
+            string option,
+            JObject rawObject,
+            string[] defaultBuiltInInclude,
+            string[] defaultBuiltInExclude)
+        {
+            if (sourceBasePath == null)
+            {
+                throw new ArgumentNullException(nameof(sourceBasePath));
+            }
+
+            if (option == null)
+            {
+                throw new ArgumentNullException(nameof(option));
+            }
+
+            if (rawObject == null)
+            {
+                throw new ArgumentNullException(nameof(rawObject));
+            }
+
+            SourceBasePath = sourceBasePath;
+            Option = option;
+            var token = rawObject.Value<JToken>(option);
+            if (token.Type != JTokenType.Object)
+            {
+                IncludePatterns = new List<string>(ExtractValues(token));
+            }
+            else
+            {
+                IncludePatterns = CreateCollection(
+                    sourceBasePath, "include", ExtractValues(token.Value<JToken>("include")), literalPath: false);
```

Comment:
The old code in `PatternsCollectionHelper` also uses this method. And that doesn't always extract values the same way. This can be refactored when we decide to remove that code.